run_pytype.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
```

Comment:
This is quite a lot of code, is this shared with xsconsole and other python projects? Would be good to add a comment where the "canonical" source for this lives (in case we improve it/fix it)

Reply:
Yes, it comes from https://github.com/xenserver/python-libs/blob/master/run-pytype.py

Yes, we should turn it into a shared pre-commit hook repository (pre-commit will fetch it then when installing its hooks - will do so locally when we run pre-commit and in CI too!).

Then, we can configure it also in .pre-commit-config.yaml and in every python project for CI as well.

Reply:
We've used this repository in the past to store project-wide CI-related settings (e.g. version numbers) https://github.com/xapi-project/xs-opam/blob/master/tools/xs-opam-ci.env
Not sure if thats the best place to store the python CI related settings, but it is one possibility

Reply:
Probably best to coordinate with @psafont on how to best share/set up a python CI, he knows a lot more about Python than I do.

Reply:
Maybe code sharing could be done as a custom Github action?  Note [there is an existing one](https://github.com/theahura/pytype-action) (did not compare the 2 approaches, though)

Reply:
Suggest to move this in a separated repo for pre-commit, just like https://github.com/hoefling/pre-commit-pytype/tree/master.

So that we can reuse this hook without duplicating run_pytype.py.

Reply:
> Maybe code sharing could be done as a custom Github action?  Note [there is an existing one](https://github.com/theahura/pytype-action) (did not compare the 2 approaches, though)

That one does not work it and it hasn't been updated in two years.

An additional problem with it is that it designed to run only as a Github acation, and has no configuration to run it from the command line, locally.

While it is possible to run a simple limited GH workflow locally inside Docker/Podman using act, it's not what I prefer to use because of its limitations and every developer who wants to test it needs to commit and push for each test, or setup act and Docker. 

But the not working and not being updated side is the actual problem.

The pytyoe runner I created can produce very nice output on GitHub in the Overview page of the Action and is configurable in more ways than is possible with pytype.cfg alone, in pyprojectl.toml.

As it also runs locally and can be even used in a pre-push and pre-commit hook, mine leads to a much smoother workflow.

I see mine therefore as vastly superior.

Reply:
> I see mine therefore as vastly superior.

Can it be packaged so it's easily installed using pip / pipx ? This would allow for code sharing as well as make it easily usable in any environment